ios/engine/KMEI/KeymanEngineTests/QueryPackageVersionTests.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  QueryPackageVersionTests.swift
+//  KeymanEngineTests
+//
+//  Created by Joshua Horton on 6/25/20.
+//  Copyright © 2020 SIL International. All rights reserved.
+//
+
+import XCTest
+@testable import KeymanEngine
+
+class QueryPackageVersionTests: XCTestCase {
+  let PACKAGE_HOST_DOMAIN = "https://downloads.keyman.com/"
+  /**
+   * A rigorous test of our package-version query code that runs against the actual, live api.keyman.com endpoint.
+   */
+  func testFetch() throws {
+    let fullIDs = [TestUtils.Keyboards.khmer_angkor.fullID,
+                   TestUtils.Keyboards.sil_euro_latin.fullID,
+                   TestUtils.LexicalModels.mtnt.fullID]
+    let expectation = XCTestExpectation(description: "Package-version query results received")
+    Queries.PackageVersion.fetch(for: fullIDs) { results, error in
+      XCTAssertNil(error)
+      XCTAssertNotNil(results)
+
+      if let results = results {
+        XCTAssertNotNil(results.keyboards)
+        XCTAssertNotNil(results.models)
+
+        if let keyboardVersions = results.keyboards, let modelVersions = results.models {
+          XCTAssertEqual(keyboardVersions.count, 2)
+          XCTAssertEqual(modelVersions.count, 1)
+
+          keyboardVersions.forEach { entry in
+            if let error = entry.value as? Queries.PackageVersion.ResultError {
+              XCTFail(String(describing: error))
+            }
+          }
+
+          if let khmer_angkor = keyboardVersions["khmer_angkor"] as? Queries.PackageVersion.ResultEntry {
+            XCTAssertGreaterThanOrEqual(Version(khmer_angkor.version)!,
+                                        Version(TestUtils.Keyboards.khmer_angkor.version)!)
+            XCTAssertTrue(khmer_angkor.packageURL.hasPrefix(self.PACKAGE_HOST_DOMAIN))
+            XCTAssertTrue(khmer_angkor.packageURL.hasSuffix("khmer_angkor.kmp"))
```

Comment:
Similarly, we shouldn't care what the url is. For example, in the future, we may well have a query string appended -- that would break this test but not the download.

Reply:
How then would you suggest testing that the query is returning meaningful data?  Without _at least_ suffix tests, it's kind of hard to test against the `kmp` field whatsoever, aside from saying "not an empty string".

Reply:
Also... chances are that if the query did return an appended query string, something _might_ break elsewhere within iOS - it makes filename assumptions in various places.  If naught else, it would indicate to double-check that the change in API returns doesn't actually break anything.